pkg/controllers/registryclient/controller.go

```diff
@@ -141,7 +175,9 @@ func (r *RegistryClientController) syncHandler(key string) error {
 	if exists {
 		// RemoteRegistry hasn't changed, do nothing
 		if registriesEqual(remoteRegistry, cachedRegistryClient.remoteRegistry) {
-			return nil
+			// If the control flow reached this point, this means the generation
+			// numbers did not match.
+			return r.updateStatus(remoteRegistry)
 		}
```

Comment:
I believe the Generation number only increments on changes to the spec on the RemoteRegistry. Since this code existed previously as an optimization to prevent redialing again (that is, it's not strictly required for correctness), I thought it would be fine to keep it.

In any case, I'm thinking a potential condition where this might occur is a situation where the RemoteRegistry changed from `A -> B -> A` , thereby changing the generation number. If the controller missed the B update, then all it would see is the generation number change.